pilot/pkg/proxy/envoy/v1/watcher.go

```diff
@@ -117,8 +117,9 @@ func (w *watcher) Reload() {
 // retrieveAZ will only run once and then exit because AZ won't change over a proxy's lifecycle
 // it has to use a reload due to limitations with envoy (az has to be passed in as a flag)
 func (w *watcher) retrieveAZ(ctx context.Context, delay time.Duration, retries int) {
-	if w.config.AvailabilityZone == "" {
-		log.Info("Availability zone not set, proxy will default to not using zone aware routing. To manually override use the --availabilityZone flag.")
+	if w.role.Type == "mixer" || w.role.Type == "pilot" {
```

Comment:
How about a flag ? Maybe off by default - with the new kube-inject it is easier to modify 
the template, and we should recommend that AZ is set by inject when possible, and enabled
when needed (and pilot has this info - I'm still not clear how does pilot know the AZ)

Reply:
I'm not sure I follow. The AZ can't be injected in Kube because the IP and AZ aren't known until after the pod has been created.

We have an issue open against envoy to configure zone aware routing natively via the xDS APIs so this retrieval is just a stop-gap solution until Envoy implements that feature.

Pilot retrieves the AZ but looking up which node the pods is on and then looking at the node labels for the region and zone labels. In Consul, it uses the datacenter field for the given service instance.